service/dap/server.go

```diff
@@ -92,15 +92,20 @@ import (
 // wrap-up and exit.
 type Server struct {
 	// config is all the information necessary to start the debugger and server.
-	config *service.Config
+	config *Config
 	// listener is used to accept the client connection.
 	listener net.Listener
-	// stopTriggered is closed when the server is Stop()-ed.
-	stopTriggered chan struct{}
-	// reader is used to read requests from the connection.
-	reader *bufio.Reader
-	// log is used for structured logging.
-	log *logrus.Entry
+	// session is the debug session that comes with an client connection.
+	session   *Session
```

Comment:
A bit weird that Server has a config but also a session that also has a config. Is it just to save keystrokes? Could the session object just be an embed?

Reply:
I was kind of modeling after rpc1/2 that also take configs. 
What exactly are you suggesting? Drop "session", so Session is embedded into Server and config field is no longer needed? In this design session is only created set if there is an open connection.

Reply:
Ok. I see now.